### PR TITLE
Fix connection leak(too many open file issue)

### PR DESCRIPTION
**Pull request by Niveathika92**
## Purpose
Fix connection leak in following scenarios,
   1. Deploy and Undeploy siddhi app containing http source - Proper closing of the connection
   2. Creating a connection only once in HTTP sink - Initially, it created for every message
Fixes https://github.com/wso2-extensions/siddhi-io-http/issues/92

## Documentation
N/A as it is a bug fix

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 1]
```diff
@@ -606,6 +606,16 @@ public void disconnect() {
...
             log.info("Server connector for url " + publisherURL + " disconnected.");
         }
+
+        if (httpConnectorFactory != null) {
+            try {
+                httpConnectorFactory.shutdown();
+                httpConnectorFactory = null;
+            } catch (InterruptedException e) {
+                log.info("Failed to shutdown the http connection factory while shutting down the siddhi app " +
+                        siddhiAppContext);
```
Comment (minudika) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: `siddhiAppContext` should be `siddhiAppContext.getName()`

Comment (Niveathika92) [reply component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: Fixed in 2490e34

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 2]
```diff
@@ -606,6 +606,16 @@ public void disconnect() {
             clientConnector = null;
             log.info("Server connector for url " + publisherURL + " disconnected.");
         }
+
+        if (httpConnectorFactory != null) {
+            try {
+                httpConnectorFactory.shutdown();
+                httpConnectorFactory = null;
+            } catch (InterruptedException e) {
+                log.info("Failed to shutdown the http connection factory while shutting down the siddhi app " +
```
Comment (AnuGayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: Isn't this actually an error log? and don't we need to add the 'e'?

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 3]
```diff
@@ -181,6 +181,11 @@ protected synchronized void initBootstrapConfigIfFirst(ConfigReader sourceConfig
     protected void clearBootstrapConfigIfLast() {
         synchronized (this) {
             if ((this.sourceListenersMap.isEmpty()) && (httpConnectorFactory != null)) {
+                try {
+                    httpConnectorFactory.shutdown();
+                } catch (InterruptedException e) {
+                    log.error("Failed to shutdown http connector factory when removing last HTTP source.");
```
Comment (AnuGayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: shall we add e here as well?

**Pull request merged**